### PR TITLE
wmic_server support

### PR DESCRIPTION
**Pull request by markdueck1**
Requires additional wmic_server to be installed / configured.

### CODE REVIEW

File: lib/NMISNG/WMI.pm [hunk 1]
```diff
@@ -181,163 +183,214 @@ sub _run_query
...
-
-		# -A format is badly documented. smbclient manpage has a little bit of info
-		# however, unclear if that password can be quoted or contain spaces or the like...
+		require Mojo::UserAgent;
+		my $headers = {"Content-type" => 'application/json', Accept => 'application/json'};
+		my $client  = Mojo::UserAgent->new();
+
+		my $url     = Mojo::URL->new( $self->{wmic_server_location} );
+		my $id = $self->{username};
+		my $token = 'MYSECRETUSERACCESSTOKEN1'; # // $self->{password};
```
Comment (jngreenwood) [inline lib/NMISNG/WMI.pm]: who sets this token and does the user need to be able to set it?

File: lib/NMISNG/WMI.pm [hunk 2]
```diff
@@ -181,163 +183,214 @@ sub _run_query
...
+		my $host = $self->{host};
+		my $namespace = "root/cimv2";
+		my $post_json = { id => $id, token => $token, host => $host, query => $query, namespace => $namespace };
+
+		my $res;
+		# connection errors die so the request needs to be wrapped in a try
+		try {
+			$res = $client->post($url => $headers => json => $post_json )->result;
```
Comment (jngreenwood) [inline lib/NMISNG/WMI.pm:207]: do we not want to use the Mojo::UserAgent error handling?

File: lib/NMISNG/WMI.pm (hunk 1 above)
Comment (markdueck1) [reply lib/NMISNG/WMI.pm]: That's me forgetting I hard coded it, I'll uncomment. The token comes from the password setting, the username is the "identity" which then matches up inside the wmic_server and resolves to the box's username/password

File: lib/NMISNG/WMI.pm (hunk 2 above)
Comment (markdueck1) [reply lib/NMISNG/WMI.pm:207]: We do further down.
From the docs: # Fine grained response handling (dies on connection errors), so without a try or eval the whole worker dies an things go bad for the whole system

**Pull request merged**